synapse/handlers/set_password.py

```diff
@@ -33,9 +33,7 @@ class SetPasswordHandler(BaseHandler):
     def __init__(self, hs: "HomeServer"):
         super().__init__(hs)
         self._auth_handler = hs.get_auth_handler()
-        device_handler = hs.get_device_handler()
-        assert isinstance(device_handler, DeviceHandler)
-        self._device_handler = device_handler
+        self._device_handler = hs.get_device_handler()
```

Comment:
So something was trying to construct this on a worker? Can we make it so that that doesn't happen?

Reply:
I'll look back into it!